src/Compilers/Core/MSBuildTask/Vbc.cs

```diff
@@ -294,6 +294,19 @@ internal void MovePdbFileIfNecessary(string outputAssembly)
                 return;
             }
 
+            // Design-time builds should not have side effects.
```

Comment:
It seems like this is fallout based on how the Execute method is being overwritten. Perhaps a better override can be provided where something only gets called if the build actually, well, built?